modules/data.remote/R/remote_process.R

```diff
@@ -0,0 +1,703 @@
+        write_raw_start <- req_start
+        write_raw_end <- req_end
+        write_pro_start <- req_start
+        write_pro_end <- req_end
+        stage_get_data <- TRUE
+        raw_merge <- FALSE
+        existing_raw_file_path = NULL
+        stage_process_data <- TRUE
+        pro_merge <- FALSE
+        existing_pro_file_path = NULL
+        flag <- 1
+      }
+    } else if (nrow(raw_check <-
+                    PEcAn.DB::db.query(
+                      sprintf(
+                        "SELECT inputs.id, inputs.site_id, inputs.name, inputs.start_date, inputs.end_date, dbfiles.file_path FROM inputs INNER JOIN dbfiles ON inputs.name=dbfiles.file_name AND inputs.name LIKE '%s%%';",
+                        raw_file_name
+                      ),
+                      dbcon
+                    )) == 1) {
+      # if only raw data is requested
+      datalist <-
+        set_stage(raw_check, req_start, req_end, stage_get_data)
+      start <- as.character(datalist[[1]])
+      end <- as.character(datalist[[2]])
+      stage_get_data <- datalist[[3]]
+      raw_merge <- datalist[[4]]
+      write_raw_start <- datalist[[5]]
+      write_raw_end <- datalist[[6]]
+      stage_process_data <- FALSE
+      if(as.character(write_raw_start) == "dont write" && as.character(write_raw_end) == "dont write"){
+        raw_id <- raw_check$id
+        raw_path <- raw_check$file_path
+      }
+      if (raw_merge == TRUE) {
+        existing_raw_file_path <- raw_check$file_path
+      } else{
+        existing_raw_file_path <- NULL
+      }
+      existing_pro_file_path <- NULL
+    } else{
+      # no data of requested type exists
+      PEcAn.logger::logger.info("Requested data does not exist in the DB, retrieving for the first time")
+      flag <- 1
+      start <- req_start
+      end <- req_end
+      if (!is.null(out_get_data)) {
+        stage_get_data <- TRUE
+        write_raw_start <- req_start
+        write_raw_end <- req_end
+        raw_merge <- FALSE
+        existing_raw_file_path = NULL
+      }
+      if (!is.null(out_process_data)) {
+        stage_process_data <- TRUE
+        write_pro_start <- req_start
+        write_pro_end <- req_end
+        pro_merge <- FALSE
+        process_file_name <- NULL
+        existing_pro_file_path <- NULL
+        flag <- 1
+      }
+    }
+    
+  } else{
+    # db is completely empty for the given siteid
+    PEcAn.logger::logger.info("DB is completely empty for this site")
+    flag <- 1
+    start <- req_start
+    end <- req_end
+    stage_get_data <- TRUE
+    write_raw_start <- req_start
+    write_raw_end <- req_end
+    raw_merge <- FALSE
+    existing_raw_file_path = NULL
+    if (!is.null(out_process_data)) {
+      stage_process_data <- TRUE
+      write_pro_start <- req_start
+      write_pro_end <- req_end
+      pro_merge <- FALSE
+      existing_pro_file_path <- NULL
+    }
+  }
+  
+  outdir = file.path(outdir, paste(source, "site", siteid_short, sep = "_"))
```

Comment:
this shouldn't be the workflow directory (they tend to be less stable), instead use this
```suggestion
  outdir <- file.path(settings$database$dbfiles, paste(source, "site", siteid_short, sep = "_"))
```